### PR TITLE
MULE-19908: set correct query params in redirect URI

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

**Comment by eze210** [comment]
--validate

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java [hunk 1]
```diff
@@ -129,6 +129,9 @@
 
   public static final String CUSTOM_MAX_HTTP_PACKET_HEADER_SIZE = SYSTEM_PROPERTY_PREFIX + "http.client.headerSectionSize";
 
+  private static final String ENABLE_MULE_REDIRECT_PROPERTY_NAME = SYSTEM_PROPERTY_PREFIX + "http.EnableMuleRedirect";
```
Comment (fsgonz) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java:132]: use camelCase for thesystem property part.

Comment (fsgonz) [reply src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java:132]: Add an explanation for this property.

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java [hunk 2]
```diff
@@ -475,7 +478,7 @@ private void handleRedirectAsync(HttpRequest request, HttpResponse response, Htt
   protected Request createGrizzlyRequest(HttpRequest request, HttpRequestOptions options)
       throws IOException {
     RequestBuilder reqBuilder = createRequestBuilder(request, options, builder -> {
-      builder.setFollowRedirects(false);
+      builder.setFollowRedirects(!enableMuleRedirect && options.isFollowsRedirect());
```
Comment (fsgonz) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java:481]: Does this mean that now, by default we are delegating the redirect logic to grizzly as before MULE-19622? 
Explain this in a comment for the system property.

File: src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    when(originalRequest.getEntity()).thenReturn(originalRequestEntity);
+
+    when(response.getHeaders()).thenReturn(responseHeaders);
+    when(response.getStatusCode()).thenReturn(301);
+    responseHeaders.put(LOCATION, "http://redirecthost/redirectPath");
+  }
+
+  @Test
+  public void redirectedRequestWithHostUsesQueryParamsFromResponseLocation() {
+    responseHeaders.removeAll(LOCATION);
```
Comment (fsgonz) [inline src/test/java/org/mule/service/http/impl/service/util/RedirectUtilsTestCase.java:64]: duplicate code. Create a method with the appropriate parameters.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java (hunk 2 above)
Comment (fsgonz) [reply src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java:483]: The answer is yes as indicated in the comment for the system property.

**Review by fsgonz — APPROVED** [review]

**Review by fsgonz — DISMISSED** [review]

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by jcgrethe** [comment]
--validate

**Comment by anosenzo** [comment]
--validate

**Comment by jcgrethe** [comment]
--validate

**Comment by jcgrethe** [comment]
--validate

**Review by fsgonz — APPROVED** [review]

**Review by chris-ebert — COMMENTED** [review]
Per HTTP 1.1 spec, 302 should use the request method. If a client needs it to change to GET, we should have a flag we can set for them to keep the non-spec behavior. We should also ask them to fix it.

**Pull request merged**